Sail/src/Sail/graphics/geometry/Transform.cpp

```diff
@@ -259,14 +259,22 @@ glm::mat4 Transform::getMatrix() {
 		updateMatrix();
 		m_parentUpdated = false;
 	}
+}
+
+glm::mat4 Transform::getMatrixWithUpdate() {
+	prepareMatrix();
+	return m_transformMatrix;
+}
+
 
+glm::mat4 Transform::getMatrixWithoutUpdate() const {
 	return m_transformMatrix;
 }
 
 glm::mat4 Transform::getRenderMatrix(float alpha) {
 	// If data hasn't changed use the CPU side transformMatrix
```

Comment:
should these three methods really return a mat4 copy or can they be changed to `const glm::mat4&` ?

Reply:
has been changed to `const glm::mat4` now